### PR TITLE
Prepare v0.24.0 release

### PR DESCRIPTION
**Pull request by lukpueh**
- bump version in setup.py
- add changelog entry

### CODE REVIEW

**Comment by lukpueh** [comment]
Please review @theupdateframework/python-tuf-maintainers, @adityasaky, @SantiagoTorres

File: CHANGELOG.md
Comment (trishankatdatadog) [inline CHANGELOG.md:8]:
```suggestion
* Use max salt lengths in RSA PSS signature creation & automatically verify previous/new sigs (#422)
```

**Comment by adityasaky** [comment]
OOTL: does this include anything that'll break in-toto / py-tuf?

**Comment by trishankatdatadog** [comment]
@yzhan289 could we please help test that this prerelease works for us?

**Comment by lukpueh** [comment]
> OOTL: does this include anything that'll break in-toto / py-tuf?

Tests pass for both against this release and there are no apparent breaking changes, so I'd say no.

**Review by mnm678 — APPROVED** [review]

**Comment by yzhan289** [comment]
Hey @lukpueh,

We found that this release of securesystemslib wouldn't be backwards compatible with previous versions. We found this by using v0.23.0 to verify signatures signed by v0.24.0.

When signing in v0.24.0, the `salt_length` is `padding.PSS.MAX_LENGTH`. It expects to see `padding.PSS.AUTO` as `salt_length` when verifying, but v0.23.0 uses `digest_obj.algorithm.digest_size` as the `salt_length` when verifying. 

Salt length PR: https://github.com/secure-systems-lab/securesystemslib/commit/1811500fd673346f1621a5db6330765352106039

cc @trishankatdatadog

**Comment by trishankatdatadog** [comment]
> When signing in v0.24.0, the `salt_length` is `padding.PSS.MAX_LENGTH`. It expects to see `padding.PSS.AUTO` as `salt_length` when verifying, but v0.23.0 uses `digest_obj.algorithm.digest_size` as the `salt_length` when verifying.

More accurately: when signing in ==v0.24.0, the salt_length is padding.PSS.MAX_LENGTH. However, <v0.24.0 uses digest_obj.algorithm.digest_size as the salt_length when verifying.

**Comment by trishankatdatadog** [comment]
This was unanticipated because we didn't realise we were still using securesystemslib (via in-toto) to sign some things on behalf of machines. So, what we should do, is make the creation of signatures backwards-compatible by providing options to create RSA PSS signatures with either the auto or hash salt length. Then, we have to patch python-in-toto to recognise this option.

**Comment by SantiagoTorres** [comment]
For the record, this also manifests in funny ways through GHA s well...

**Comment by adityasaky** [comment]
> So, what we should do, is make the creation of signatures backwards-compatible by providing options to create RSA PSS signatures with either the auto or hash salt length. Then, we have to patch python-in-toto to recognise this option.

Do you reckon you have the bandwidth for the sslib patch? We could put this patch and some DSSE stuff out as 0.25.0 and jump straight to it in in-toto / python-tuf.

**Comment by adityasaky** [comment]
I'm guessing it's also okay to hold off on packaging 0.24.0 for Arch or Debian or elsewhere (sslib is now packaged in openbsd ports https://marc.info/?l=openbsd-ports-cvs&m=166128655601083&w=2 as well)

**Comment by trishankatdatadog** [comment]
Yes, sending off a PR today (pair-programmed with @yzhan289)

**Comment by trishankatdatadog** [comment]
> Yes, sending off a PR today (pair-programmed with @yzhan289)

Please see https://github.com/secure-systems-lab/securesystemslib/pull/431

**Pull request merged**
